amazon-kinesis-client/src/test/java/software/amazon/kinesis/common/StreamIdentifierTest.java

```diff
@@ -0,0 +1,94 @@
+package software.amazon.kinesis.common;
+
+import com.google.common.base.Joiner;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import software.amazon.awssdk.arns.Arn;
+import software.amazon.awssdk.regions.Region;
+
+import java.util.Optional;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.when;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(StreamARNUtil.class)
+public class StreamIdentifierTest {
+    private static final String streamName = "streamName";
+    private static final Region kinesisRegion = Region.US_WEST_1;
+    private static final String accountId = "111111111111";
+    private static final String epoch = "1680616058";
+
+    @Test
+    public void testSingleStreamInstanceWithName() {
+        StreamIdentifier actualStreamIdentifier = StreamIdentifier.singleStreamInstance(streamName);
+        Assert.assertFalse(actualStreamIdentifier.streamCreationEpochOptional().isPresent());
+        Assert.assertFalse(actualStreamIdentifier.accountIdOptional().isPresent());
+        Assert.assertFalse(actualStreamIdentifier.streamARNOptional().isPresent());
+        Assert.assertEquals(streamName, actualStreamIdentifier.streamName());
+    }
+
+    @Test
+    public void testSingleStreamInstanceWithNameAndRegion() {
+        Optional<Arn> arn = Optional.of(Arn.builder().partition("aws").service("kinesis")
+                .region(kinesisRegion.toString()).accountId("123").resource("stream/" + streamName).build());
+        mockStatic(StreamARNUtil.class);
+        when(StreamARNUtil.getStreamARN(eq(streamName), eq(kinesisRegion))).thenReturn(arn);
+        StreamIdentifier actualStreamIdentifier = StreamIdentifier.singleStreamInstance(streamName, kinesisRegion);
+        Assert.assertFalse(actualStreamIdentifier.streamCreationEpochOptional().isPresent());
+        Assert.assertFalse(actualStreamIdentifier.accountIdOptional().isPresent());
```

Comment:
Is this correct? When a provided region yields an Arn, should we set the account id to match that from the Arn (if it was going to be otherwise empty)?

Anotherwards, can/should this be `assertEquals(Optional.of("123"), aSI.accountIdOptional())` ?